src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java

```diff
@@ -115,6 +121,30 @@ protected void actionPerformed(final GuiButton btn) {
         }
     }
 
+    @Override
+    protected void mouseClicked(final int xCoord, final int yCoord, final int btn) {
+        if (hoveredStack != null && isShiftKeyDown()) {
+            NBTTagCompound data = Platform.openNbtData(hoveredStack);
+            WorldCoord blockPos2 = new WorldCoord(
+                    (int) mc.thePlayer.posX,
+                    (int) mc.thePlayer.posY,
+                    (int) mc.thePlayer.posZ);
+            for (DimensionalCoord blockPos : DimensionalCoord.readAsListFromNBT(data)) {
+                BlockPosHighlighter.highlightBlock(
+                        blockPos,
+                        System.currentTimeMillis() + 500 * WorldCoord.getTaxicabDistance(blockPos, blockPos2));
```

Comment:
For 100 blocks distance, this will show it for 50 sec. Put an upper limit on the time